src/helpers/transport/query.rs

```diff
@@ -69,6 +64,46 @@ impl RouteParams<RouteId, NoQueryId, NoStep> for &QueryConfig {
     }
 }
 
+impl From<&QueryConfig> for GatewayConfig {
+    fn from(_value: &QueryConfig) -> Self {
+        // TODO: pick the correct value for active and test it
+        Self::default()
+    }
+}
+
+impl QueryConfig {
+    /// Initialize new query configuration.
+    ///
+    /// ## Errors
+    /// If query size is too large or 0.
+    ///
+    /// ## Panics
+    /// if query size does not fit into a pointer-sized type.
+    pub fn new<S>(
+        query_type: QueryType,
+        field_type: FieldType,
+        size: S,
+    ) -> Result<Self, QueryConfigError>
+    where
+        <S as TryInto<usize>>::Error: Debug,
+        S: TryInto<usize> + TryInto<u32>,
+    {
+        let sz: usize = size.try_into().expect("query size fits into machine word");
+        let sz = u32::try_from(sz)
+            .and_then(NonZeroU32::try_from)
+            .map_err(|_| QueryConfigError::BadQuerySize {
+                actual_size: sz,
+                max_size: MAX_QUERY_SIZE,
```

Comment:
Are you relying on `MAX_QUERY_SIZE` being equal to `NonZeroU32::MAX`? 

... and I just saw that it is not.  We'll need a new type for this, with a `TryFrom` implementation that doesn't accept 2<sup>31</sup>, as this does.